GsaGHTests/1_BaseParameters/0_Model/GsaModelTest.cs

```diff
@@ -130,5 +131,78 @@ public void TestModelLengthUnit() {
 
       Assert.Equal(LengthUnit.Foot, UnitMapping.GetUnit(m.ApiModel.UiUnits().LengthLarge));
     }
+
+    [Theory]
+    [InlineData(0)]
+    [InlineData(1)]
+    [InlineData(2)]
+    public void ModalAnalysisTaskAreCopiedInDuplicateModel(int methodId) {
+      var original = new GsaModel();
+      int numberOfMode = 5;
+      // Task
+      var massOption = new MassOption(ModalMassOption.LumpMassAtNode, 1);
+      var additionalMassDerivedFromLoads = new AdditionalMassDerivedFromLoads(
+          "L1",
+          Direction.Z,
+          1
+      );
+      var ModalDamping = new ModalDamping(0.5);
+      var modalDynamicTaskParameter = new ModalDynamicTaskParameter(
+          ModeCalculationMethod(methodId, numberOfMode),
+          massOption,
+          additionalMassDerivedFromLoads,
+          ModalDamping
+      );
+
+      int taskId = original.ApiModel.AddAnalysisTask(
+          AnalysisTaskFactory.CreateModalDynamicAnalysisTask(
+              "task1",
+              modalDynamicTaskParameter
+          )
+      );
+      for (int mode = 1; mode <= numberOfMode; mode++) {
+        original.ApiModel.AddAnalysisCaseToTask(taskId, "test case", mode);
+      }
+      System.Collections.ObjectModel.ReadOnlyDictionary<int, AnalysisTask> taskIn = original.ApiModel.AnalysisTasks();
+
+      //assemble model and get task
+      var analysis = new GsaAnalysis();
+      foreach (KeyValuePair<int, AnalysisTask> analysisTask in taskIn) {
+        analysis.AnalysisTasks.Add(new GsaAnalysisTask(analysisTask.Key, analysisTask.Value, original.ApiModel));
+      }
+      var assembly = new ModelAssembly(new GsaModel(), null, null, null, null, null, analysis,
+        LengthUnit.Meter, Length.Zero, false, null);
+      var assembled = new GsaModel() {
+        ApiModel = assembly.GetModel()
+      };
+      System.Collections.ObjectModel.ReadOnlyDictionary<int, AnalysisTask> taskOut = assembled.ApiModel.AnalysisTasks();
+
+      Assert.Equal(taskIn.Count, taskIn.Count);
+      foreach (int key in taskIn.Keys) {
+        Assert.Equal(taskIn[key].Name, taskOut[key].Name);
+        foreach (int caseId in taskIn[key].Cases) {
+          Assert.Equal(assembled.ApiModel.AnalysisCaseName(caseId), original.ApiModel.AnalysisCaseName(caseId));
+          Assert.Equal(assembled.ApiModel.AnalysisCaseDescription(caseId), original.ApiModel.AnalysisCaseDescription(caseId));
+        }
+      }
+    }
```

Comment:
this is too big

Can you create one test per exist condition of your new method?

You need 3 tests where the `numberOfDynamicMode` is larger that 0 but comes from the 3 different type of `modelCalculationStrategy` that it should have the `analysisCaseToTask` being added like that `_model.AddAnalysisCaseToTask(task.Id, ca.Name, mode);` and another case where the numberOfDynamicMode is Zero and the analysisCastToTask has been added through `_model.AddAnalysisCaseToTask(task.Id, ca.Name, ca.Definition);`